skyportal/handlers/api/gcn_aliases.py

```diff
@@ -0,0 +1,251 @@
+import arrow
+import requests
+import re
+from ..base import BaseHandler
+from baselayer.app.access import permissions
+from ...models import GcnEvent
+from astropy.time import Time
+
+
+def get_tach_event_id(dateobs, tags):
+    date = dateobs.split("T", 1)[0]
+    url = (
+        "https://heasarc.gsfc.nasa.gov/wsgi-scripts/tach/gcn_v2/tach.wsgi/graphql_fast"
+    )
+
+    payload = {
+        "query": f'''{{
+              allEventDetails (  date1:\"{date}\" date2:\"{date}\" ) {{
+                totalCount
+                pageInfo{{
+                  hasNextPage
+                  hasPreviousPage
+                  startCursor
+                  endCursor
+                }}
+                edges {{
+                  node {{
+                    id
+                    id_
+                    event
+                    evttype
+                    trigger
+                    ra
+                    raHms
+                    dec
+                    decDms
+                    error
+                    date
+                    circulars
+                    notices
+                    circnum
+                    }}
+                  }}
+                }}
+            }}'''
+    }
+    headers = {
+        "Content-Type": "application/json",
+        "Origin": "https://heasarc.gsfc.nasa.gov",
+    }
+
+    response = requests.request("POST", url, json=payload, headers=headers)
+    data = response.json()
+    if response.status_code != 200:
+        return None
+    if response.json()["data"]["allEventDetails"]["totalCount"] == 0:
+        return None
+
+    events = data["data"]["allEventDetails"]["edges"]
+    event_id = None
+    for event in events:
+        trigger = event["node"]["trigger"]
+        if trigger is not None:
+            try:
+                trigger = Time(Time(trigger, precision=0).iso).datetime.strftime(
+                    "%Y-%m-%dT%H:%M:%S"
+                )
+            except Exception:
+                continue
+            if trigger == dateobs:
+                event_id = event["node"]["id_"]
+
+    if event_id is None:
+        event_ids = []
+        for event in events:
+            if event["node"]["date"] == date and event["node"]["evttype"] in tags:
+                event_ids.append(event["node"]["id_"])
+        if len(event_ids) != 1:
+            # multiple events on the same day, cant figure out the right one. We could look at some circular and notices, event types.... to find which one is the right one.
```

Comment:
Is there a plan to add some code to make a decision between multiple GCNs in case there are a few with the same date/type?